tap_chargebee/streams/base.py

```diff
@@ -24,9 +24,11 @@ def generate_catalog(self):
 
         metadata = {
             "selected": self.SELECTED,
+            "replication-method": self.REPLICATION_METHOD,
+            "replication-key": self.REPLICATION_KEY,
             "inclusion": self.INCLUSION,
-            "valid-replication-keys": self.VALID_REPLICATION_KEYS,
             "selected-by-default": self.SELECTED_BY_DEFAULT,
+            "valid-replication-keys": self.VALID_REPLICATION_KEYS,
             "schema-name": self.TABLE
         }
```

Comment:
I think this is still going to run into errors if its run in Stitch.

The `selected`, `replication-method`, and `replication-key` metadata keys are `non-discoverable`, meaning they should only be read by a tap, and are written by systems outside the tap (such as Stitch).  The `schema-name` is only intended for database taps (this was an error in the documentation, which I just fixed).  

The replication method can be set during discovery using `forced-replication-method` metadata, and the key_properties can be set during discovery using `table-key-properties` metadata.  

So I think this chunk should look something like:
```
metadata = {
   "forced-replication-method": self.REPLICATION_METHOD,
   "valid-replication-keys": self.BOOKMARK_PROPERTIES,
   "inclusion": self.INCLUSION,
   "selected-by-default": self.SELECTED_BY_DEFAULT,
   "table-key-properties": self.KEY_PROPERTIES
}
```

Reply:
Also, I can't figure out how to comment on it, but below on line 46, I think you also want to mark any bookmark props as `automatic` as well, so that line should read:
```
if field_name in self.KEY_PROPERTIES or field_name in self.BOOKMARK_PROPERTIES:
```